src/mcts/search.cc

```diff
@@ -150,17 +153,25 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
       cache_->GetSize() * 1000LL / std::max(cache_->GetCapacity(), 1);
   uci_info_.nps =
       uci_info_.time ? (total_playouts_ * 1000 / uci_info_.time) : 0;
-  uci_info_.score = 290.680623072 * tan(1.548090806 * best_move_edge_.GetQ(0));
-  uci_info_.pv.clear();
+  uci_info_.comment.clear();
 
-  bool flip = played_history_.IsBlackToMove();
-  for (auto iter = best_move_edge_; iter;
-       iter = GetBestChildNoTemperature(iter.node()), flip = !flip) {
-    uci_info_.pv.push_back(iter.GetMove(flip));
-    if (!iter.node()) break;  // Last edge was dangling, cannot continue.
+  // Handle UCI multipv. Loop until we are out of moves or kMultiPv
+  std::vector<EdgeAndNode> sorted_edges = GetDescChildrenNoTemperature(root_node_);
+  for (int pv_idx = 0;
+       pv_idx < std::min(kMultiPv, static_cast<int>(sorted_edges.size()));
+       ++pv_idx) {
+    EdgeAndNode edge = sorted_edges[pv_idx];
+    bool is_black = played_history_.IsBlackToMove();
+    uci_info_.score = EdgeAndNode::GetCpScore(edge);
```

Comment:
I believe I was doing this (just with a static method) and I don't believe the node method is always non-null because it was crashing on me. I decided to go with EdgeAndNode for that reason.

Reply:
Without checking for nulls I get a crash like this 

```
2018-08-07 17:30:30.397<--1:Creating backend [cudnn]...
2018-08-07 17:30:31.042<--1:readyok
2018-08-07 17:30:31.042<--1:readyok
2018-08-07 17:30:31.042<--1:readyok
2018-08-07 17:30:31.080<--1:info depth 2 seldepth 2 time 19 nodes 3 score cp 18 hashfull 0 nps 157 pv c2c4 e7e6 multipv 1
2018-08-07 17:30:31.088<--1:info depth 2 seldepth 2 time 19 nodes 3 score cp 10 hashfull 0 nps 157 pv e2e3 g7g6 multipv 2
2018-08-07 17:30:31.112<--1:BytesRead > 0
2018-08-07 17:30:31.113<--1:-- LiesThread.Execute terminated --
```
With edge.node()->GetCpScore(); sometimes